pkg/secrethub/credentials/setup_code.go

```diff
@@ -0,0 +1,25 @@
+package credentials
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/secrethub/secrethub-go/internals/auth"
+)
+
+type SetupCode struct {
+	code string
+}
+
+func NewSetupCode(code string) *SetupCode {
+	return &SetupCode{
+		code: code,
+	}
+}
+
+// Authenticate authenticates the given request with a setup code, by providing the "SetupCode" tag and the setup code
+// in the "Authorization" header.
+func (s *SetupCode) Authenticate(r *http.Request) error {
+	r.Header.Set("Authorization", fmt.Sprintf("%s-%s %s", auth.AuthHeaderVersionV1, "SetupCode", s.code))
+	return nil
+}
```

Comment:
https://github.com/secrethub/secrethub-go/blob/9cc951048891ec17e56d4130cba3bf24fe9c1257/internals/auth/signature.go#L91
`auth.signer.Authenticate` seems to be the method used to set the `Authorization` header in requests, but it cannot be used in this case since it uses a different format for the `Authorization` header (`%s-%s %s:%s` instead of `%s-%s %s`).

The simplest way I found to implement this was to set the `Authorization` header manually like this :point_up: 

Is there a way to improve this code?

Reply:
Actually this is exactly how it was intended to be done. `auth.signer.Authenticate` is only intended to be used for authentication by signing the request. Setup codes use another form of authentication, so they implement their own function.